hcloud/schema.go

```diff
@@ -1,1271 +1,370 @@
-			IPv4: LoadBalancerPublicNetIPv4{
-				IP:     net.ParseIP(s.PublicNet.IPv4.IP),
-				DNSPtr: s.PublicNet.IPv4.DNSPtr,
-			},
-			IPv6: LoadBalancerPublicNetIPv6{
-				IP:     net.ParseIP(s.PublicNet.IPv6.IP),
-				DNSPtr: s.PublicNet.IPv6.DNSPtr,
-			},
-		},
-		Location:         LocationFromSchema(s.Location),
-		LoadBalancerType: LoadBalancerTypeFromSchema(s.LoadBalancerType),
-		Algorithm:        LoadBalancerAlgorithm{Type: LoadBalancerAlgorithmType(s.Algorithm.Type)},
-		Protection: LoadBalancerProtection{
-			Delete: s.Protection.Delete,
-		},
-		Labels:          map[string]string{},
-		Created:         s.Created,
-		IncludedTraffic: s.IncludedTraffic,
-	}
-	for _, privateNet := range s.PrivateNet {
-		l.PrivateNet = append(l.PrivateNet, LoadBalancerPrivateNet{
-			Network: &Network{ID: privateNet.Network},
-			IP:      net.ParseIP(privateNet.IP),
-		})
-	}
-	if s.OutgoingTraffic != nil {
-		l.OutgoingTraffic = *s.OutgoingTraffic
-	}
-	if s.IngoingTraffic != nil {
-		l.IngoingTraffic = *s.IngoingTraffic
-	}
-	for _, service := range s.Services {
-		l.Services = append(l.Services, LoadBalancerServiceFromSchema(service))
-	}
-	for _, target := range s.Targets {
-		l.Targets = append(l.Targets, LoadBalancerTargetFromSchema(target))
-	}
-	for key, value := range s.Labels {
-		l.Labels[key] = value
-	}
-	return l
+	return c.LoadBalancerFromSchema(s)
+}
+
+// SchemaFromLoadBalancer converts a LoadBalancer to a schema.LoadBalancer.
+func SchemaFromLoadBalancer(l *LoadBalancer) schema.LoadBalancer {
+	return c.SchemaFromLoadBalancer(l)
 }
 
 // LoadBalancerServiceFromSchema converts a schema.LoadBalancerService to a LoadBalancerService.
 func LoadBalancerServiceFromSchema(s schema.LoadBalancerService) LoadBalancerService {
-	ls := LoadBalancerService{
-		Protocol:        LoadBalancerServiceProtocol(s.Protocol),
-		ListenPort:      s.ListenPort,
-		DestinationPort: s.DestinationPort,
-		Proxyprotocol:   s.Proxyprotocol,
-		HealthCheck:     LoadBalancerServiceHealthCheckFromSchema(s.HealthCheck),
-	}
-	if s.HTTP != nil {
-		ls.HTTP = LoadBalancerServiceHTTP{
-			CookieName:     s.HTTP.CookieName,
-			CookieLifetime: time.Duration(s.HTTP.CookieLifetime) * time.Second,
-			RedirectHTTP:   s.HTTP.RedirectHTTP,
-			StickySessions: s.HTTP.StickySessions,
-		}
-		for _, certificateID := range s.HTTP.Certificates {
-			ls.HTTP.Certificates = append(ls.HTTP.Certificates, &Certificate{ID: certificateID})
-		}
-	}
-	return ls
+	return c.LoadBalancerServiceFromSchema(s)
+}
+
+// SchemaFromLoadBalancerService converts a LoadBalancerService to a schema.LoadBalancerService.
+func SchemaFromLoadBalancerService(l LoadBalancerService) schema.LoadBalancerService {
+	return c.SchemaFromLoadBalancerService(l)
 }
 
 // LoadBalancerServiceHealthCheckFromSchema converts a schema.LoadBalancerServiceHealthCheck to a LoadBalancerServiceHealthCheck.
 func LoadBalancerServiceHealthCheckFromSchema(s *schema.LoadBalancerServiceHealthCheck) LoadBalancerServiceHealthCheck {
```

Comment:
Tests for those are already implicitly included in `TestLoadBalancerServiceSchema` and `TestLoadBalancerTargetSchema`. Do you think it would make sense to extract them and test them seperately?

Reply:
I ran the test suite with coverage and these 4 methods were showing up as not covered. Just ran the tests you mentioned and the actual converter methods in `converterImpl` are covered :)